src/Tribe/Admin/Notice/Install_Event_Tickets.php

```diff
@@ -0,0 +1,436 @@
+		}
+
+		$vars    = $_REQUEST;
+		$success = false;
+
+		if ( 'install' === $vars['request'] ) {
+			$success = $this->action_install( $vars );
+		} elseif ( 'activate' === $vars['request'] ) {
+			$success = $this->action_activate( $vars );
+		}
+
+		if ( false === $success ) {
+			$install_url = wp_nonce_url(
+				self_admin_url( 'update.php?action=install-plugin&plugin=' . $vars['slug'] ),
+				'install-plugin_' . $vars['slug']
+			);
+			$message     = sprintf(
+				/* Translators: %1$s - opening link tag, %2$s - closing link tag. */
+				__( 'There was an error and plugin could not be installed, %1$splease install manually%2$s.', 'the-events-calendar' ),
+				'<a href="' . esc_url( $install_url ) . '">',
+				'</a>',
+			);
+
+			wp_send_json_error( [ 'message' => wpautop( $message ) ] );
+		} else {
+			wp_send_json_success( [ 'message' => __( 'Success.', 'the-events-calendar' ) ] );
+		}
+	}
+
+	/**
+	 * Action to install & activate plugin.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $vars The AJAX vars.
+	 *
+	 * @return bool $success True if the plugin was successfully installed and activated.
+	 */
+	public function action_install( $vars ): bool {
+		if ( ! function_exists( 'plugins_api' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+		}
+		if ( ! class_exists( 'WP_Upgrader' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+		}
+
+		$api = plugins_api(
+			'plugin_information',
+			[
+				'slug'   => $vars['slug'],
+				'fields' => [
+					'short_description' => false,
+					'sections'          => false,
+					'requires'          => false,
+					'rating'            => false,
+					'ratings'           => false,
+					'downloaded'        => false,
+					'last_updated'      => false,
+					'added'             => false,
+					'tags'              => false,
+					'compatibility'     => false,
+					'homepage'          => false,
+					'donate_link'       => false,
+				],
+			]
+		);
+		if ( ! is_wp_error( $api ) ) {
+			$upgrader  = new \Plugin_Upgrader( new \WP_Ajax_Upgrader_Skin() );
+			$installed = $upgrader->install( $api->download_link );
+
+			if ( $installed ) {
+				$activate = activate_plugin( $this->get_plugin_path( $vars['slug'] ), '', false, true );
+				$success  = ! is_wp_error( $activate );
+			} else {
+				$success = false;
+			}
+		} else {
+			$success = false;
+		}
+
+		return $success;
+	}
+
+	/**
+	 * Action to activate the plugin.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $vars The AJAX vars.
+	 *
+	 * @return bool $success True if the plugin was successfully activated.
+	 */
+	public function action_activate( $vars ) {
+		if ( ! $this->is_installed( $vars['slug'] ) ) {
+			return $this->action_install( $vars );
+		}
+
+		if ( $this->is_active( $vars['slug'] ) ) {
+			return true;
+		}
+
+		$activate = activate_plugin( $this->get_plugin_path( $vars['slug'] ), '', false, true );
+
+		return ! is_wp_error( $activate );
+	}
```

Comment:
These methods should not live in this class and must be tested in some capacity.

Reply:
These are reliant on WP core functions that are already tested in WP. Where would you move these? As they are only applicable (at this point) to the functionality related to this notice.

Reply:
I would move those into Common, but honestly, it's fine if it lives here now, and we can move the whole class later to Common so we can leverage this for other plugins' behavior of installing other plugins on the go.